test/helpers/kubectl.go

```diff
@@ -486,6 +494,16 @@ func (kub *Kubectl) CiliumEndpointPolicyVersion(pod string) map[string]int64 {
 
 // CiliumExec runs cmd in the specified Cilium pod.
 func (kub *Kubectl) CiliumExec(pod string, cmd string) *CmdRes {
+	if config.CiliumTestConfig.Developer {
```

Comment:
Having these sorts of execution paths will make maintenance of this code difficult IMO. We should really move to using one base VM image for all development. It will make the code much cleaner.